apps/core/tasks/auto_upgrade/runner.py

```
@@ -0,0 +1,290 @@
+                ),
+            )
+            return None, False
+
+    command = args
+    run_kwargs: dict[str, object] = {}
+
+    if os.name == "nt":
+        run_kwargs["shell"] = True
+
+    try:
+        subprocess.run(command, cwd=base_dir, check=True, **run_kwargs)
```

Ruff (0.14.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)